app/celery/scheduled_tasks.py

```diff
@@ -141,3 +149,36 @@ def send_daily_performance_platform_stats():
             email_sent_count,
             'day'
         )
+
+
+@notify_celery.task(name='switch-current-sms-provider-on-slow-delivery')
+@statsd(namespace="tasks")
+def switch_current_sms_provider_on_slow_delivery():
+    """
+    Switch providers if there are at least two slow delivery notifications (more than four minutes)
+    in the last ten minutes. Search from the time we last switched to the current provider.
+    """
+    functional_test_provider_service_id = current_app.config.get('FUNCTIONAL_TEST_PROVIDER_SERVICE_ID')
+    functional_test_provider_template_id = current_app.config.get('FUNCTIONAL_TEST_PROVIDER_SMS_TEMPLATE_ID')
+
+    if functional_test_provider_service_id and functional_test_provider_template_id:
+        current_provider = get_current_provider('sms')
+        slow_delivery_notifications = get_count_of_slow_delivery_sms_notifications_for_provider(
+            provider=current_provider.identifier,
+            threshold=2,
+            created_at=current_provider.updated_at,
+            sent_at=datetime.utcnow() - timedelta(minutes=10),
```

Comment:
<-----10 minutes ago-----provider updated (7 mins ago)------now-->

This works well as will also ensure that we only check in a marginal window (from provider updated to now which is 3 mins) if we updated the provider only a couple of minutes ago.